### DISCUSSION TITLE
Unable to install Ruby 2.3.8 on mac M1. (macOS 13.0.1 using ruby-build 20221124)

### DISCUSSION THREAD

**Discussion by peromax** [rbenv/ruby-build#2098] [Build failures] [answered]
I trying install ruby 2.3.8 using openssl@1.0 [(1.0.2u)](https://github.com/rbenv/homebrew-tap/pull/2#issuecomment-1047430606) because ruby version < 2.4

After run:
`RUBY_CONFIGURE_OPTS="--with-openssl-dir=$(brew --prefix openssl@1.0)" RUBY_CFLAGS='-Wno-error=implicit-function-declaration'  rbenv install 2.3.8`

Had error:
```
BUILD FAILED (macOS 13.0.1 using ruby-build 20221124)

Inspect or clean up the working tree at /var/folders/26/blxb__vx6n544yd6ys1tvn640000gn/T/ruby-build.20221129040626.20489.QcutHZ
Results logged to /var/folders/26/blxb__vx6n544yd6ys1tvn640000gn/T/ruby-build.20221129040626.20489.log

Last 10 log lines:

Search tcl.h.
Search tk.h.Can't find "tk.h".

Can't find proper Tcl/Tk libraries. So, can't make tcltklib.so which is required by Ruby/Tk.
If you have Tcl/Tk libraries on your environment, you may be able to use them with configure options (see ext/tk/README.tcltklib).
At present, Tcl/Tk8.6 is not supported. Although you can try to use Tcl/Tk8.6 with configure options, it will not work correctly. I recommend you to use Tcl/Tk8.5 or 8.4.
Failed to configure tk. It will not be installed.
Failed to configure tk/tkutil. It will not be installed.
configuring zlib
```

One of the error from log file:
```
...
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [.ext/-darwin22/enc/big5.bundle] Error 1
...
```

Attached full log file:
[ruby-build.20221129040626.20489.log](https://github.com/rbenv/ruby-build/files/10107141/ruby-build.20221129040626.20489.log)

Additional info about environment: 

- Apple Silicon M1, 
- Xcode (14.1), 
- Command_Line_Tools_for_Xcode_14.1 

If someone had similar issue please help! Thanks in advance 🙏

**Comment by peromax** [comment, marked as answer]
I tried frum, and it's works. Maybe this will help someone:
`RUBY_CFLAGS='-DUSE_FFI_CLOSURE_ALLOC -Wno-error=implicit-function-declaration'  frum install 2.3.8 --with-openssl-dir=$(brew --prefix openssl@1.0) --enable-shared`

On macOs Ventura.

**Reply by aubri61** [reply]
I tried this with ventura 13.1 but didn't work..

this kind of error came out. can you please give me the clue?


> BUILD FAILED (macOS 13.1 using ruby-build 20221225)
> 
> Inspect or clean up the working tree at /var/folders/j0/4ncpc9m90tg_c7d24lc1kvnr0000gn/T/ruby-build.20230116112549.2689.Ll53rF
> Results logged to /var/folders/j0/4ncpc9m90tg_c7d24lc1kvnr0000gn/T/ruby-build.20230116112549.2689.log
> 
> Last 10 log lines:
> installing default syslog libraries
> compiling date_parse.c
> linking shared-object gdbm.bundle
> compiling date_strftime.c
> installing digest libraries
> compiling date_strptime.c
> linking shared-object objspace.bundle
> linking shared-object nkf.bundle
> linking shared-object date_core.bundle
> make: *** [build-ext] Error 2

**Comment by hsbt** [comment]
Unfortunately, there is no way to build Ruby 2.3 with macOS 13.x Ventura or Xcode14. Ruby core team only support to build Ruby 2.7-3.1+ with them.

**Reply by peromax** [reply]
Not really, as I did manage to install ruby 2.3.8, but had to use `frum` instead of `rbenv`
<img width="477" alt="Screenshot 2022-11-29 at 23 52 26" src="https://user-images.githubusercontent.com/988665/204577652-64347ce9-a7fe-41fa-988f-b6e32ce705cd.png">

**Reply by hsbt** [reply]
feel free to use it with your responsibility. Ventura have a `fork` issue for the old Ruby versions. FYI: https://github.com/ruby/ruby/pull/6426

**Reply by sfahado** [reply]
Hi @peromax, could you please share your experience on how you successfully installed it on your local machine? I'm using M1 and Sonoma, but I'm encountering issues.

**Reply by monfresh** [reply]
@sfahado If you're specifically looking to install Ruby 2.3.8, then your best bet will probably be to use my paid [Ruby on Mac](https://www.rubyonmac.dev/) product (the Ultimate version). That's because Ruby versions older than 2.4 require OpenSSL 1.0, which you can no longer install with Homebrew. Alternatively, you could try to compile OpenSSL 1.0 yourself, and then install Ruby 2.3.8.

**Comment by manjula-balamurugan** [comment]
`rbenv install 2.5.1`


rbenv: /Users/manjula/.rbenv/versions/2.5.1 already exists
continue with installation? (y/N) y
To follow progress, use 'tail -f /var/folders/th/y8wwbp616svgczl6ky8b8rlw0000gq/T/ruby-build.20221202122628.16683.log' or pass --verbose
Downloading openssl-1.1.1s.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/c5ac01e760ee6ff0dab61d6b2bbd30146724d063eb322180c6f18a6f74e4b6aa
Installing openssl-1.1.1s...
Installed openssl-1.1.1s to /Users/manjula/.rbenv/versions/2.5.1

Downloading ruby-2.5.1.tar.bz2...
-> https://cache.ruby-lang.org/pub/ruby/2.5/ruby-2.5.1.tar.bz2
Installing ruby-2.5.1...

WARNING: ruby-2.5.1 is past its end of life and is now unsupported.
It no longer receives bug fixes or critical security updates.

ruby-build: using readline from homebrew
ruby-build: using gmp from homebrew

BUILD FAILED (macOS 13.0.1 using ruby-build 20221124)

Inspect or clean up the working tree at /var/folders/th/y8wwbp616svgczl6ky8b8rlw0000gq/T/ruby-build.20221202122628.16683.1li0z6
Results logged to /var/folders/th/y8wwbp616svgczl6ky8b8rlw0000gq/T/ruby-build.20221202122628.16683.log

Last 10 log lines:
../.././include/ruby/ruby.h:1755:56: note: expanded from macro 'rb_intern'
        __extension__ (RUBY_CONST_ID_CACHE((ID), (str))) : \
                                                       ^
1332 warnings generated.
linking shared-object objspace.bundle
1 warning generated.
linking shared-object psych.bundle
436 warnings generated.
linking shared-object date_core.bundle
make: *** [build-ext] Error 2



Env : Mac OS 13 
Unable to install ruby 2.5.1 which is the required version ,Please help how to resolve this

**Reply by mislav** [reply]
Hi, Ruby 2.5 is End-Of-Life and we cannot provide support for compiling it anymore, especially not on macOS Ventura.

I would suggest updating the Ruby version in your app to 2.7, or better to Ruby 3.1.

**Reply by AdityaTiwari2102** [reply]
@mislav - I'm getting the same error with Ruby 2.7.2 and 3.1.2.

**Reply by mislav** [reply]
@AdityaTiwari2102 Please first try latest Ruby 2.7.x and 3.1.x since they can contain build fixes.

**Reply by AdityaTiwari2102** [reply]
@mislav - Latest Ruby 2.7.x is installed successfully but 3.1.x is still throwing the same error.

**Reply by mislav** [reply]
@AdityaTiwari2102 And what is the error for you? Note that the OP from this thread hasn't actually shared their error message nor full build logs.

**Reply by AdityaTiwari2102** [reply]
@mislav - I don't have the error handy currently, but I was able to get it installed using the 1st workaround mentioned [here](https://github.com/rbenv/ruby-build/issues/2073#issuecomment-1335651657).
